dataRetriever/topicSender/topicResolverSender.go

```diff
@@ -30,13 +31,15 @@ func NewTopicResolverSender(arg ArgTopicResolverSender) (*topicResolverSender, e
 
 // Send is used to send an array buffer to a connected peer
 // It is used when replying to a request
-func (trs *topicResolverSender) Send(buff []byte, peer core.PeerID) error {
-	// TODO[Sorin]: add a new field on MessageP2P for the network the message should be on
-	if trs.fullArchiveMessenger.IsConnected(peer) {
-		return trs.sendToConnectedPeer(trs.topicName, buff, peer, trs.fullArchiveMessenger, fullArchiveNetwork, trs.fullArchivePreferredPeersHolderHandler)
+func (trs *topicResolverSender) Send(buff []byte, peer core.PeerID, destination p2p.MessageHandler) error {
+	switch destination.Type() {
+	case p2p.RegularMessageHandler:
```

Comment:
this switch defeats the purpose of providing the p2p.MessageHandler because we need to provide also the peers holder handler. Please put back the TODO to think of a solution to this

Reply:
removed it as discussed, now the check for preferred is done on both main and full archive holders